src/k_point/k_point.hpp

```diff
@@ -60,6 +60,10 @@ class K_point
 
     std::unique_ptr<spfft::Transform> spfft_transform_;
 
+    /// A float type fft transform to supprt fp32 calculation
+#ifdef USE_FP32
+    std::unique_ptr<spfft::TransformFloat> spfft_transformFloat_;
```

Comment:
Shouldn't it go into K-point PR?

Reply:
It was indeed a previous commit that intent not to template the k_point class. I remove it now, but keep in mind that the templated hamiltonian class for this is not ready for a float type instantiation as the spfft driver is incorrect there

Reply:
Got it! We will first merge the K_point PR and then resolve the conflicts here.